### PR TITLE
Add rollback/commit streaming DistSQL implementation

### PR DESCRIPTION
**Pull request by azexcy**
Related #22500

Changes proposed in this pull request:
  - Add drop streaming DistSQL implementation
  - Extraction methods for reuse

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/impl/CDCJobAPI.java [hunk 1]
```diff
@@ -218,6 +220,45 @@ protected String marshalJobIdLeftPart(final PipelineJobId pipelineJobId) {
...
+     *
+     * @param jobId    job id
+     * @param disabled disabled
+     */
+    public void updateJobConfigurationDisabled(final String jobId, final boolean disabled) {
+        JobConfigurationPOJO jobConfigPOJO = getElasticJobConfigPOJO(jobId);
+        updateJobConfigurationDisabled(jobConfigPOJO, disabled);
+    }
+    
+    private void updateJobConfigurationDisabled(final JobConfigurationPOJO jobConfigPOJO, final boolean disabled) {
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/impl/CDCJobAPI.java:250]: Could we combine these 2 methods?

File: kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/impl/CDCJobAPI.java [hunk 2]
```diff
@@ -218,6 +220,45 @@ protected String marshalJobIdLeftPart(final PipelineJobId pipelineJobId) {
...
+        // TODO, ensure that there is only one consumer at a time, job config disable may not be updated when the program is forced to close
+        jobConfigPOJO.setDisabled(disabled);
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/impl/CDCJobAPI.java:252]: Could this TODO resolved this time? What's the plan if it's not

File: kernel/data-pipeline/distsql/handler/src/main/java/org/apache/shardingsphere/cdc/distsql/handler/update/DropStreamingUpdater.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * Drop streaming updater.
+ */
+public final class DropStreamingUpdater implements RALUpdater<DropStreamingStatement> {
+    
+    private final CDCJobAPI jobAPI = new CDCJobAPI();
+    
+    @Override
+    public void executeUpdate(final String databaseName, final DropStreamingStatement sqlStatement) throws SQLException {
+        jobAPI.rollback(sqlStatement.getJobId());
+    }
```
Comment (sandynz) [inline kernel/data-pipeline/distsql/handler/src/main/java/org/apache/shardingsphere/cdc/distsql/handler/update/DropStreamingUpdater.java:36]: Could we keep `commit` and `rollback` but not `drop`?

File: test/it/parser/src/main/resources/sql/supported/ral/cdc.xml [hunk 4]
```diff
@@ -19,4 +19,5 @@
 <sql-cases>
     <sql-case id="show-streaming-list" value="SHOW STREAMING LIST;" db-types="ShardingSphere"/>
     <sql-case id="show-streaming-status" value="SHOW STREAMING STATUS 123;" db-types="ShardingSphere"/>
+    <sql-case id="drop-streaming" value="DROP STREAMING 123;" db-types="ShardingSphere"/>
```
Comment (sandynz) [inline test/it/parser/src/main/resources/sql/supported/ral/cdc.xml:22]: Could we add new DistSQL test in E2E?

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/impl/CDCJobAPI.java (hunk 2 above)
Comment (azexcy) [reply kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/impl/CDCJobAPI.java:252]: I will removed

File: proxy/frontend/core/src/main/java/org/apache/shardingsphere/proxy/frontend/netty/CDCChannelInboundHandler.java [hunk 5]
```diff
@@ -128,8 +128,8 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
             case START_STREAMING:
                 processStartStreamingRequest(ctx, request, connectionContext);
                 break;
-            case DROP_STREAMING:
-                processDropStreamingRequest(ctx, request, connectionContext);
+            case ROLLBACK_STREAMING:
+                processRollbackStreamingRequest(ctx, request, connectionContext);
```
Comment (sandynz) [inline proxy/frontend/core/src/main/java/org/apache/shardingsphere/proxy/frontend/netty/CDCChannelInboundHandler.java:132]: Seems COMMIT is missing

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandler.java [hunk 6]
```diff
@@ -41,14 +40,9 @@
...
 import org.apache.shardingsphere.data.pipeline.cdc.util.CDCSchemaTableUtils;
 import org.apache.shardingsphere.data.pipeline.cdc.util.CDCTableRuleUtils;
-import org.apache.shardingsphere.data.pipeline.core.api.PipelineAPIFactory;
 import org.apache.shardingsphere.data.pipeline.core.context.PipelineContextManager;
 import org.apache.shardingsphere.data.pipeline.core.exception.job.PipelineJobNotFoundException;
 import org.apache.shardingsphere.data.pipeline.core.job.PipelineJobCenter;
-import org.apache.shardingsphere.data.pipeline.core.job.PipelineJobIdUtils;
-import org.apache.shardingsphere.elasticjob.infra.pojo.JobConfigurationPOJO;
-import org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrap;
-import org.apache.shardingsphere.elasticjob.lite.lifecycle.api.JobConfigurationAPI;
```
Comment (sandynz) [inline proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/cdc/CDCBackendHandler.java:51]: Could we move CDCBackendHandler to CDC module

**Review by sandynz — CHANGES_REQUESTED** [review]

**Review by sandynz — APPROVED** [review]

**Pull request merged**